build.rs

```diff
@@ -40,5 +48,60 @@ fn main() {
         println!("cargo:rustc-link-lib=static=chck-xdg");
         println!("cargo:rustc-link-lib=static=wlc-protos");
         println!("cargo:rustc-link-lib=static=wlc");
+    } else if let Err(s) = find() {
+        let _ = writeln!(io::stderr(), "{}", s);
+        process::exit(1);
+    }
+}
+
+fn find() -> Result<(), Error> {
+
+    let package_name = "wlc";
+    let shared_libs = ["wlc"];
+
+    // https://github.com/way-cooler/rust-wlc#rust-wlc currently
+    // (i.e. commit 97e0d5816fe7ec97dae20196d1e25c550f60f08f) suggests
+    // version 0.0.7.
+    let version = if cfg!(feature = "0.0.8") {
+        "0.0.8"
+    } else if cfg!(feature = "0.0.7") {
+        "0.0.7"
+    } else if cfg!(feature = "0.0.6") {
+        "0.0.6"
+    } else {
+        "0.0.5"
+    };
+
+    // Is this Windoze-specific? Do we need to carry this along on Linux with
+    // Wayland at all?
+    let target = env::var("TARGET").unwrap();
+    let hardcode_shared_libs = target.contains("windows");
+
+    let mut config = Config::new();
+    config.atleast_version(version);
+
+    if hardcode_shared_libs {
+        config.cargo_metadata(false);
+    }
+
+    match config.probe(package_name) {
+        Ok(library) => {
+            if hardcode_shared_libs {
+                for lib_ in shared_libs.iter() {
+                    println!("cargo:rustc-link-lib=dylib={}", lib_);
+                }
+                for path in library.link_paths.iter() {
+                    println!("cargo:rustc-link-search=native={}", path.to_str().unwrap());
+                }
```

Comment:
These nice loops seem to be doing what that ugly multi-line `println!` thing I have in the main function doing. Can we not just add those libraries I have defined (e.g `wayland-client`, `X11-xcb`, etc.) into the library list and generate them if `CARGO_FEATURE_STATIC_WLC` is set? If not, and we need to have these two branches, it's fine.